include/lsst/afw/geom/detail/wcsUtils.h

```diff
@@ -0,0 +1,165 @@
+/*
+ * LSST Data Management System
+ * Copyright 2017 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#ifndef LSST_AFW_GEOM_DETAILS_WCSUTILS_H
+#define LSST_AFW_GEOM_DETAILS_WCSUTILS_H
+
+#include <memory>
+#include <string>
+
+#include "Eigen/Core"
+
+#include "lsst/daf/base.h"
+#include "lsst/afw/geom/Point.h"
+#include "lsst/afw/geom/SkyWcs.h"
+
+namespace lsst {
+namespace afw {
+namespace geom {
+namespace detail {
+
+/**
+ * @internal Define a trivial named WCS that maps the lower left corner pixel of an image to its parent
+ * coordinate.
+ *
+ * Sets the following keywords:
+ * - CRPIX[12]<wcsName> = 1
+ * - CRVAL[12]<wcsName> = xy0
+ * - CTYPE[12]<wcsName> = "LINEAR"
+ * - CUNIT[12]<wcsName> = "PIXEL"
+ *
+ * @param wcsName WCS name, e.g. "A"
+ * @param xy0  Parent coordinate of lower left corner pixel of image
+ */
+std::shared_ptr<daf::base::PropertyList> createTrivialWcsAsPropertySet(std::string const& wcsName,
+                                                                       geom::Point2I const& xy0);
+
+/**
+ * @internal Delete metadata for a named WCS
+ *
+ * Delete keywords created by createTrivialWcsAsPropertySet plus CD keywords and WCSAXES.
+ * Missing entries are silently ignored.
+ *
+ * @param[in,out] metadata The metadata
+ * @param[in] wcsName The WCS to search, e.g. "A"
+ */
+void deleteBasicWcsMetadata(daf::base::PropertySet& metadata, std::string const& wcsName);
+
+/**
+ * Read a CD matrix from FITS WCS metadata
+ *
+ * The elements of the returned matrix are in degrees
+ *
+ * @throws pex::exceptions::InvalidParameterError if no CD matrix coefficients found
+ * (missing coefficients are set to 0, as usual, but they cannot all be missing).
+ */
+Eigen::Matrix2d getCdMatrixFromMetadata(daf::base::PropertySet& metadata);
```

Comment:
Because the CD matrix is always called CD. I am not aware of another FITS set of keywords formatted the same as CD, though there may well be some. SIP terms are 0-indexed and have a leading underscore before the first index.